docs/getting-started.md

```diff
@@ -0,0 +1,463 @@
+
+1. Create a file in `tests` called `test-hello-ostree.sh`.
+
+2. Add the following to `test-hello-ostree.sh`:
+
+    ```bash
+    set -euo pipefail           # Ensure the test will not silently fail
+
+    . $(dirname $0)/libtest.sh  # Make libtest.sh functions available
+
+    echo "1..1"                 # Declare which test is being run out of how many
+
+    pushd ${test_tmpdir}
+
+    ${CMD_PREFIX} ostree hello-ostree > hello-output.txt
+    assert_file_has_content hello-output.txt "Hello OSTree!"
+
+    popd
+
+    echo "ok hello ostree"      # Indicate test success
+    ```
+
+    Many tests require a fake repository setting up (as most OSTree commands require `--repo` to be specified). See `test-pull-depth.sh` for an example of this setup.
+
+3. Configure automake to include `test-hello-ostree.sh` in the build, by adding an entry in `Makefile-tests.am` under `_installed_or_uninstalled_test_scripts`:
+
+    ```bash
+    tests/test-hello-ostree.sh \
+    ```
+
+4. Make sure `test-hello-ostree.sh` has executable permissions!
+
+    ```bash
+    chmod +x tests/test-hello-ostree.sh
+    ```
+
+5. Run the test:
+
+    ```
+    make check TESTS="test/test-hello-ostree.sh
+    ```
+
+    Multiple tests can be specified: `make check TESTS="test1 test2 ..."`. To run all tests, use `make check`.
+
+    Hopefully, the test passes! The following will be printed:
+
+    ```bash
+    PASS: tests/test-hello-ostree.sh 1 hello ostree
+    ============================================================================
+    Testsuite summary for libostree 2018.8
+    ============================================================================
+    # TOTAL: 1
+    # PASS:  1
+    # SKIP:  0
+    # XFAIL: 0
+    # FAIL:  0
+    # XPASS: 0
+    # ERROR: 0
+    ============================================================================
+    ```
+
+### Submitting a Patch
+
+After you have committed your changes and tested, you are ready to submit your patch!
+
+You should make sure your commits are placed on top of the latest changes from `upstream/master`:
+
+```bash
+$ git pull --rebase upstream master
+```
+
+To submit your patch, open a pull request from your forked repository. Most often, you'll be merging into `ostree:master` from `<username>:<branch name>`.
+
+If some of your changes are complete and you would like feedback, you may also open a pull request that has WIP (Work In Progress) in the title.
+
+Before a pull request is considered merge ready, your commit messages should fall within the specified guideline. See [Commit message style](CONTRIBUTING.md#commit-message-style).
+
+See [CONTRIBUTING.md](https://github.com/ostreedev/ostree/blob/master/docs/CONTRIBUTING.md#submitting-patches) for information on squashing commits, and alternative options to submit patches.
+
+### Returning Workflow
+
+When returning to make a new patch, do the following to update your fork with the latest changes to master:
+
+```bash
+git fetch upstream
```

Comment:
For this step, I usually do

```
$ git checkout master
$ git pull upstream master
```

Not sure if one is better than the other, though.

Reply:
Thinking about it, the section could include two use cases like so:

If creating a new branch:

```
$ git checkout master
$ git pull upstream master
$ git checkout -b <name-of-patch>
```

If continuing on a branch already created:

```
$ git checkout <name-of-patch>
$ git pull --rebase upstream master
```

(may as well avoid separating the fetch, and no need to `checkout -b` at the end in the case of an already existing feature branch)

@cgwalters suggested rebasing initially. I like the rebase way to update existing branches and avoid interleaving commits - and it saves master from getting interleaved commits if there were local commits made to the forked master. However rebasing is not necessary to update a forked master and branch out - merge works fine then.

Will add in the edit for the two use cases above, if that looks good.